lib/collection/property.js

```diff
@@ -170,6 +170,17 @@ _.extend(Property, /** @lends Property */ {
      * @returns {String}
      */
     replaceSubstitutions: function (str, variables) {
```

Comment:
also, please add a `// @todo: improve algorithm by doing variable level replacement cacheing` since the current algorithm is a dfs and can receive some improvements later.